event/oracle/register_oracle_event.go

```diff
@@ -20,6 +27,62 @@ func (e RegisterOracleEvent) GetEventQuery() string {
 }
 
 func (e RegisterOracleEvent) EventHandler(event ctypes.ResultEvent) error {
-	// TODO: https://github.com/medibloc/panacea-oracle/issues/3
-	panic("will be implemented in issue #3")
+	uniqueID := event.Events[oracletypes.EventTypeRegistration+"."+oracletypes.AttributeKeyUniqueID][0]
+	targetAddress := event.Events[oracletypes.EventTypeRegistration+"."+oracletypes.AttributeKeyOracleAddress][0]
+
+	msgApproveOracleRegistration, err := e.verifyAndGetMsgApproveOracleRegistration(uniqueID, targetAddress)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("new oracle registration approval info. uniqueID(%s), approverAddress(%s), targetAddress(%s)",
+		msgApproveOracleRegistration.ApproveOracleRegistration.UniqueId,
+		msgApproveOracleRegistration.ApproveOracleRegistration.ApproverOracleAddress,
+		msgApproveOracleRegistration.ApproveOracleRegistration.TargetOracleAddress,
+	)
+
+	txBuilder := panacea.NewTxBuilder(e.reactor.QueryClient())
+	txBytes, err := txBuilder.GenerateTxBytes(e.reactor.OracleAcc().GetPrivKey(), e.reactor.Config(), msgApproveOracleRegistration)
+	if err != nil {
+		return err
+	}
+
+	txHeight, txHash, err := e.reactor.BroadcastTx(txBytes)
+	if err != nil {
+		return fmt.Errorf("failed to ApproveOracleRegistration transaction for new oracle registration: %v", err)
+	} else {
+		log.Infof("succeeded to ApproveOracleRegistration transaction for new oracle registration. height(%v), hash(%s)", txHeight, txHash)
+	}
+
+	return nil
+}
+
+func (e RegisterOracleEvent) verifyAndGetMsgApproveOracleRegistration(uniqueID, targetAddress string) (*oracletypes.MsgApproveOracleRegistration, error) {
+	queryClient := e.reactor.QueryClient()
+	approverAddress := e.reactor.OracleAcc().GetAddress()
+	oraclePrivKeyBz := e.reactor.OraclePrivKey().Serialize()
+	approverUniqueID := e.reactor.EnclaveInfo().UniqueIDHex()
+
+	if uniqueID != approverUniqueID {
+		log.Infof("oracle's uniqueID does not match the requested uniqueID. expected(%s) got(%s)", approverUniqueID, uniqueID)
+	} else {
```

Comment:
If unique ID is different, why don't you return an error directly with logging?

Reply:
Ah I want to make a same error return method, if there has errors. But I think if the uniqueID is different, it should be return an error.

Reply:
Fixed in [fbd3f92](https://github.com/medibloc/panacea-oracle/pull/21/commits/fbd3f92755daaeeb436f0910f2c67a81b30fb503).